cmd/protecodeExecuteScan.go

```diff
@@ -175,13 +167,18 @@ func executeProtecodeScan(client protecode.Protecode, config *protecodeExecuteSc
 	result := client.PollForResult(productID, config.TimeoutMinutes)
 
 	jsonData, _ := json.Marshal(result)
-	filePath := filepath.Join(reportPath, "protecodescan_vulns.json")
+	filePath := filepath.Join(reportPath, scanResultFile)
 	ioutil.WriteFile(filePath, jsonData, 0644)
 
 	//check if result is ok else notify
-	if len(result.Result.Status) > 0 && result.Result.Status == "F" {
+	if len(result.Result.Status) > 0 && result.Result.Status == protecode.StatusFailed {
 		log.Entry().Fatalf("Please check the log and protecode backend for more details. URL: %v/products/%v", config.ServerURL, productID)
 	}
+
+	if config.FailOnSevereVulnerabilities && protecode.HasSevereVulnerabilities(result.Result, config.ExcludeCVEs) {
+		log.Entry().Fatalf("The product is not compliant. URL: %v/products/%v", config.ServerURL, productID)
```

Comment:
Would it not make tests easier to return an error here? (also on line 175)

Reply:
Yes, if I'm correct the best practice is to use log.Fatal only in the main function, right?

Reply:
We recommend it in Development.md. You can assert the code-path easier. Instead of having to register and unregister a fatal handler on the log package, you can simply assert an error returned from runXY().

Reply:
I would need to do more refactoring to fix this.

Reply:
No problem. I already approved anyway.

Reply:
@stippi2 I changed my mind and did the change here, can you have another look?